github_connector/models/github_repository_branch.py

```diff
@@ -126,7 +144,9 @@ def button_download_code(self):
         return self._download_code()
 
     def button_analyze_code(self):
-        return self._analyze_code()
+        res = self._analyze_code()
+        self._analysis_code()
```

Comment:
seems duplicated.

Reply:
Although seems duplicate the function is different and the functionality too (first function is related to github operations and the new function check code lines).

Reply:
sorry I missread your code.

Reply:
Well, both functions seems very similar. Why not join them together and make only one call, keeping for compatibility the previous name?

Reply:
In this case i had create othe function because in the future other addons can need to use this, but yes, maybe keep only one function in this case is a good idea.

Reply:
If you end up keeping both functions separate, I'd suggest it would be better to rename them (or at least one) to improve readability for the future.

Reply:
Looks better with the latest commit :+1:

Reply:
Finally use the original function and inside this function call other custom funcions (in the future its very possible to override this new functions)